crc32_fold.c

```diff
@@ -17,6 +17,14 @@ Z_INTERNAL void crc32_fold_copy_c(crc32_fold *crc, uint8_t *dst, const uint8_t *
     memcpy(dst, src, len);
 }
 
+Z_INTERNAL void crc32_fold_c(crc32_fold *crc, const uint8_t *src, size_t len, uint32_t init_crc) {
+    /* Note: while this is basically the same thing as the vanilla CRC function, we still need
```

Comment:
`init_crc` is unused? Is this intended...

Reply:
Nevermind I see your comment.

Reply:
Yes, init_crc is for an initial crc value.  It's used to XOR in with the first 4 bytes loaded.  Because the normal folding + copy crc didn't need to worry about this (the initial value was assumed 0), this implicitly carried through an initial crc of 0.

The new CRC is a generic CRC that can be called from anywhere with an initial CRC already specified, and in order to mix that into the new sequence of CRCs, it needed to be a function argument.  Because our surrogate for the CRC folding actually carried through the initial CRC for the C version anyhow by stashing it in the "fold" variable, this argument is technically redundant for the case where we're using this with a "folded" CRC (as in, we had a crc_fold variable established and we're rolling through CRC calculation by doing the carryless multiplications).  

What I maybe could have done instead is modify the "reset" function to take an argument that is the initial CRC value and write to to "value".  Then each folded CRC I could have maybe written code that checks this value to determine whether or not value is still set to something non-zero.  But that's also a bit hackish, and also it makes the copy + CRC code less efficient, since the same checks to established whether or not an initial CRC32 is defined need to exist in front of every load from src.